jsonschema-core.xml

```diff
@@ -903,6 +928,24 @@
                     </t>
                 </section>
             </section>
+            <section title="Dereferencing By Inclusion">
```

Comment:
As I mentioned in the previous PR (I think, or maybe on Slack), I think that if this is just to explain the difference from prior behavior, it belongs in an appendix (which would be xref'd from a brief note in the main text).  This is what I've been doing for all confusing changes, like the `dependencies` split, `$defs` vs `definitions`, and keywords that moved from validation to core.

If we do want to provide guidance on an inclusion mechanism for things like json-schema-ref-parser that want to do so to the extent it is possible, then it should describe the use with `allOf` per #523.

But I don't think there should be a dereference inclusion process as part of the standard.  Just a clear description of the behavior, and tools that want to make schema transformations that preserve that behavior can do so (for instance, I know of at least three tools for collapsing `allOf` branches into a single schema, which is not part of the spec but follows directly from the described behavior).